Issue: Refactor the installation process

bengaal opened the pull request.

Commit:
Applied Coding Standard

Applied Black Coding Standard on following files.

Commit:
inveniordmpure Renamed inveniordmpure to invenio_rdm_pure

Renamed class inveniordmpure to invenio_rdm_pure in order to
comply with standard defined by Cookiecutter Invenio Module
Template.

bengaal committed:
ext.py Removed translator instruction comment

Removed translator instruction comment, since it was unnecessary.
The comment stems from the cookiecutter module template.

bengaal committed:
config.py setup.py Sorted imports

bengaal committed:
ext.py Removed translation string

bengaal committed:
config.py Moved configuration variables

To skip the initial setup process these variables are going
to be initialized here, and then overwritten by the values in the
invenio.cfg configuration file.

bengaal committed:
rdm_host_url Removed unused variable imports

Removed unreferenced variable imports

bengaal committed:
Sorted imports

bengaal committed:
setup.py Removed rdm_host_url and rdm_records_url variables

Removed variable definitions and their references. They
get their values from the config file now.

bengaal committed:
setup.py Removed token_rdm

Globally removed token_rdm, since token validation won't be used.

bengaal committed:
setup.py Removed pure_rest_api_url variable

Removed variable, and its references. The value comes from the
configuration now.

bengaal committed:
setup.py Removed pure_api_key variable

Removed pure_api_key initialization from setup.py. The value comes
from the configuration file instead.

bengaal committed:
setup.py Removed pure_password variable

Removed pure_password variable from setup.py and other references
of this variable in the rest of the code. This value comes from
the configuration now.

bengaal committed:
setup.py Removed pure_username variable

Removed pure_username variable from setup.py and other references
of this variable in the rest of the code. This value comes from
the configuration now.

bengaal committed:
setup.py Removed email_receiver variable

Removed email_receiver variable from setup.py and other references
of this variable in the rest of the code. This value comes from
the configuration now.

bengaal committed:
setup.py Removed email_sender variable

Removed email_sender variable from setup.py and other references
of this variable in the rest of the code. This value comes from
the configuration now.

bengaal committed:
setup.py Removed email_sender_password variable

Removed email_sender_password variable from setup.py and other references
of this variable in the rest of the code. This value comes from
the configuration now.

bengaal committed:
setup.py Removed pure_rdm_user_file variable

Removed pure_rdm_user_file variable from setup.py and other references
of this variable in the rest of the code. This value isn't needed
anymore.

bengaal committed:
setup.py Removed pure_rdm_password_file variable

Removed email_sender_password variable from setup.py.
This value was never and it won't be used.

bengaal committed:
Sorted imports and applied coding standard

bengaal committed:
invenio_rdm_pure Renamed invenio_rdm_pure to InvenioRdmPure

Renamed class invenio_rdm_pure to InvenioRdmPure in order to comply
with naming convention of PEP8.

mb-wali commented on invenio_rdm_pure/cli.py:
```diff
@@ -28,6 +28,7 @@
 from docopt import docopt
 from shell_interface import ShellInterface, method_call
 from source.general_functions import check_if_directory_exists
+
```
This file runs in a separate environment rather than the module itself.
we need to refactor the ```cli.py``` file for this module. and use the [Flask.commands](https://flask.palletsprojects.com/en/1.1.x/cli/).

utnapischtim replied:
I think that should belong to another PR and be handled in #10.

I added my comment to this issue.

mb-wali commented on invenio_rdm_pure/config.py:
```diff
@@ -93,3 +93,25 @@
 RDM_RECORDS_METADATA_NAMESPACES = {
     "tug": {"@context": "https://graz.pure.elsevier.com/"}
 }
+
+# Invenio-RDM-Pure
+# =========================
+#
+INVENIO_RDM_PURE_CONFIG_PRESENT = False
+## Invenio-RDM
+## ========================
+##
+RDM_HOST_URL = "https://127.0.0.1:5000/"
+RDM_RECORD_URL = RDM_HOST_URL + "api/records/{}"
+RDM_RECORDS_URL = RDM_HOST_URL + "api/records"
+RDM_PURE_USER_EMAIL = ""
+RDM_PURE_USER_PASSWORD = ""
```
```suggestion

# Invenio-RDM-Pure
# =========================
#

INVENIO_RDM_PURE_CONFIG_PRESENT = False
"""What is it for - short comment needed."""

# Invenio-RDM
# ========================
#

RDM_HOST_URL = "https://127.0.0.1:5000/"
"""What is it for - short comment needed."""

RDM_RECORD_URL = RDM_HOST_URL + "api/records/{}"
"""What is it for - short comment needed."""

RDM_RECORDS_URL = RDM_HOST_URL + "api/records"
"""What is it for - short comment needed."""

RDM_PURE_USER_EMAIL = ""
"""What is it for - short comment needed."""

RDM_PURE_USER_PASSWORD = ""
"""What is it for - short comment needed."""
```

mb-wali commented on invenio_rdm_pure/config.py:
```diff
@@ -93,3 +93,25 @@
+RDM_RECORDS_URL = RDM_HOST_URL + "api/records"
+RDM_PURE_USER_EMAIL = ""
+RDM_PURE_USER_PASSWORD = ""
+
+## Pure
+## ========================
+##
+PURE_API_URL = ""
+PURE_API_KEY = ""
+PURE_USERNAME = ""
+PURE_PASSWORD = ""
+PURE_RESPONSIBLE_EMAIL = ""
```
```suggestion

# Pure
# ========================
#

PURE_API_URL = ""
"""What is it for - short comment needed."""

PURE_API_KEY = ""
"""What is it for - short comment needed."""

PURE_USERNAME = ""
"""What is it for - short comment needed."""

PURE_PASSWORD = ""
"""What is it for - short comment needed."""

PURE_RESPONSIBLE_EMAIL = ""
"""What is it for - short comment needed."""
```

mb-wali replied:
Also for configuration variables - it's better to use a prefix such as ```INVENIO``` or ```PURE``` to all newly added config vars. for example: ```INVENIO_PURE_PASSWORD```:  open for discussion @utnapischtim .

utnapischtim replied:
I added my comment to the issue #11 

In short, we should distinguish between between the configuration variables which belong to the repository and those which belong to the source/target system. 

For me calling the configuration variable PURE_PASSWORD makes it more clear that it is the password to connect the pure system. With INVENIO_PURE_PASSWORD it is not that clear for me.

mb-wali reviewed (changes requested):
The focus of this review was based on changes made to the module, and I left some comments, to improve that structure of the module so it will meet the inveniosoftware structure.

when going through the codes I came across some points and opened some issues.
* defining config variables
* write flask CLI

bengaal committed:
config.py Applied changes suggested in pull-request #9

Config variable disambiguations regarding variablename and
descriptions.

bengaal committed:
config.py Refined variable names and descriptions

Chose better variable names to make them more uniform and refined
their descriptions.

bengaal committed:
database_uri.py Migrated database attributes to config

The database related attributes were generated upon calling
localhost/database_uri.
Now these attributes are acquired in ext.py and added to
app.config upon execution.

bengaal committed:
config.py Added docstyle formatting
